lib/partition_setup.pm

```diff
@@ -187,22 +187,21 @@ sub addlv {
 }
 
 sub select_first_hard_disk {
-    assert_screen [qw(existing-partitions hard-disk-dev-sdb-selected)];
-    if (match_has_tag 'hard-disk-dev-sdb-selected' || get_var('SELECT_FIRST_DISK')) {
-        if (check_var('VIDEOMODE', 'text')) {
-            if (match_has_tag 'hotkey_d') {
-                send_key 'alt-d';
-            }
-            elsif (match_has_tag 'hotkey_e') {
-                send_key 'alt-e';
+    my $matched_needle = assert_screen [qw(existing-partitions hard-disk-dev-sdb-selected  hard-disk-dev-non-sda-selected)];
+    if (match_has_tag 'hard-disk-dev-non-sda-selected' || match_has_tag 'hard-disk-dev-sdb-selected' || get_var('SELECT_FIRST_DISK')) {
+        # SUT may have any number disks, only keep the first, unselect all other disks
+        foreach my $tag (@{$matched_needle->{needle}->{tags}}) {
+            if (check_var('VIDEOMODE', 'text')) {
+                if ($tag =~ /hotkey_([a-z])/) {
```

Comment:
Yes I like your suggestion actually. However may add difficulty for someone to read the code. How do you think?

Reply:
@rwx788 thanks for review, we can leave it for future improvement. I also have offline discussion with @alice-suse and going to merge it since it is in urgent, and she would care about needles because it is very rely on them.